### PR TITLE
Add --ignore-resource-errors switch to hide resource errors from output

### LINKED ISSUE #153: Option to disable "Error loading resource" error
Hi yall!

Have to say I love this project. So simple and easy to get setup!

But I was wondering, I keep getting "Error loading resource" all over my reporting. It'd be amazing to be able to disable that somehow,  there's like custom webfonts and SVG images that the `loadImages` option doesn't take care of.

Not sure if this is possible, if it isn't feel free to close.

Thanks!

### PR DESCRIPTION
**Pull request by andreialecu**
This fixes #153 by adding an --ignore-resource-errors switch to hide resource errors.

### CODE REVIEW

**Comment by nathanboktae** [comment]
I would not use `-i` but only `--ignore-resource-errors` . its verbose but this option should be rare - 404s are generally bugs unless its testing handing of these conditions.

Also a test for this please.

**Comment by andreialecu** [comment]
In the tests we're running we call an api via ajax, and the api replies
with non 200 codes. It may reply with 404 when an item is not found, which
is something we test for.

We also test for malformed input which also makes the api reply with non
200 codes.

These api calls show up in logs.

See here for an example:
https://travis-ci.org/deployd/deployd/builds/45888892
Scroll to the bottom.

I will rename it, but I'm not sure how to test it or if a test is really
relevant since the change is so minor and straight forward.

>
>

**Comment by andreialecu** [comment]
I pushed another commit where it is renamed as per your suggestion @nathanboktae 

Regarding tests, I looked a bit at them and I don't know how to test this without a web server. All the tests seem to be local right now, and resource errors only happen with remote requests. Any ideas?

**Comment by nathanboktae** [comment]
Yeah I was thinking about it more, and it is tricky as there is no depedency outside the box the build is running on, and no local server yet. Also no test for the existing functionality either.... I'll take a crack at it.

Yes thanks for removing `-i` - I just checked and mocha uses it to invert `-g`, so we'll use it that way too when we implement `-g` (#145)

**Comment by nathanboktae** [comment]
Yeah I was thinking about the test, and it is tricky to implement as we do not have a dependency outside the box for the CI build, nor a server running during tests. Also nor do we have tests for the existing functionality! I'll take a look at it later.

Also I did double-check and mocha uses `-i` to invert grep results, so we'll implement that when `-g` gets implemented (tracked in #145)

**Comment by nathanboktae** [comment]
Thought I already left this comment but I didn't release resource errors trigger for XHRs - I only thought they triggered for missing resources like scripts, images, CSS. Now it makes it a more common scenario. Maybe the shorthand could be `-I` - thoughts @metaskills ?

**Comment by metaskills** [comment]
TL;DR - Just pick anything that makes sense.

Names of args is hard :( which side do we lean too. I think there are seldom good answers and we should just be careful and cognitive. So first side (mocha) or (phantomjs) to implement the need is maybe our winner :)

**Comment by andreialecu** [comment]
By the way, a new release having this new functionality would be great so we can plug it into deployd and have our Travis logs look cleaner. :)

**Comment by nathanboktae** [comment]
You can point directly at master for the moment. I started working on the test for this and it's easy with a bad css, script, or image url, so I'll add it soon.

**Comment by michasko** [comment]
Is there a way to use this option within grunt config object with [mocha_phantomjs](https://github.com/jdcataldo/grunt-mocha-phantomjs) plugin?
E.g.:

    mocha_phantomjs: {
      all: {
        options: {
          '--ignore-resource-errors': 'true'
        }
      }
    }

Sorry if a silly question, I'm new to all this.

**Comment by dfrencham** [comment]
@michasko no mate, the '--ignore-resource-errors' option is for mocha-phantomjs only.

**Comment by nathanboktae** [comment]
Not true - we are using it in our gruntfile. remove the `--`:

```
    mocha_phantomjs: {
      all: ['tests.html'],
      options: {
        'ignore-resource-errors': true
      }
    }
```

**Comment by michasko** [comment]
@nathanboktae yes, after a lot of tries (camelCase, single dash at the beginning) I went through the code and figured it out. I forgot to update my post for others, though. Thanks!
